extensions/kubernetes-client/deployment/src/main/java/io/quarkus/kubernetes/client/deployment/KubernetesClientProcessor.java

```diff
@@ -20,14 +25,30 @@
     BuildProducer<ReflectiveClassBuildItem> reflectiveClasses;
 
     @BuildStep
-    public void process(CombinedIndexBuildItem combinedIndexBuildItem,
+    public void process(ApplicationIndexBuildItem applicationIndex, CombinedIndexBuildItem combinedIndexBuildItem,
             BuildProducer<ExtensionSslNativeSupportBuildItem> sslNativeSupport) {
         featureProducer.produce(new FeatureBuildItem(FeatureBuildItem.KUBERNETES_CLIENT));
 
+        Set<String> watchedClasses = new HashSet<>();
+        // make sure the watchers fully (and not weakly) register Kubernetes classes for reflection
+        applicationIndex.getIndex().getAllKnownImplementors(DotName.createSimple("io.fabric8.kubernetes.client.Watcher"))
+                .forEach(c -> {
+                    final MethodInfo eventReceived = c.firstMethod("eventReceived");
+                    if (eventReceived != null) {
+                        if (eventReceived.parameters().size() == 2) {
+                            watchedClasses.add(eventReceived.parameters().get(1).name().toString());
+                        }
+                    }
+                });
+        if (!watchedClasses.isEmpty()) {
+            reflectiveClasses.produce(new ReflectiveClassBuildItem(true, true, watchedClasses.toArray(new String[0])));
```

Comment:
Wouldn't it be better to make the array the size of `watchedClasses`?

Reply:
That's what I did initially, but it turns out that it's not needed - IntelliJ auto-suggests change